### PR TITLE
Allows for plugin-specific chunks sizes

### PR DESCRIPTION
**Pull request by darrylmasson**
This changes the default uncompressed chunk size from 100 MB (specified in an obscure, undocumented location) to a plugin-definable value, with new default 200 MB. This value gets stored with the metadata, but if it's missing the default is assumed. This should affect writing data much more than reading it, but the hope is that 200 MB uncompressed is a more acceptable trade between fewer, larger files (easier on the bookkeeping) and more, smaller files (easier on the end user).

### CODE REVIEW

**Comment by darrylmasson** [comment]
It's worth pointing out that having this as a user-specifiable value means there is considerable misuse potential, and so maintainers should make sure that instanced Plugins don't set this to absurd values.

**Comment by jorana** [comment]
@darrylmasson as Travis is complaining, you want to export via __all__. See https://github.com/xenonnt/straxen/blob/master/straxen/common.py#L20 for an example

**Comment by darrylmasson** [comment]
Let's see if this fixes it

File: strax/chunk.py [hunk 1]
```diff
@@ -31,7 +34,8 @@ def __init__(self,
                  run_id,
                  start,
                  end,
-                 data):
+                 data,
+                 target_size=default_chunk_size):
```
Comment (jorana) [inline strax/chunk.py:39]: Can we be smart about the naming such that no-one accidentally assumes that the target_size is in MB (which would result in saying target_size=200 which would then result in infinite files in the ./strax_data directory where one tests a plugin prior to pushing it to github).

For example, how about target_size_mb (and multiply later by 1000).

Comment (darrylmasson) [reply strax/chunk.py:39]: We could, or we explicitly document it as bytes. We should not be held accountable if someone decides that they don't need to familiarize themselves with the documentation before changing attributes.

Comment (jorana) [reply strax/chunk.py:39]: Agree, people should read the documentation. However, all numbers that can be used here will be in x MB so I think having it in some int between 10-3000 would make it quite readable and easier to spot if someone is doing something that he/she shouldn't. Another example is if you cat the metadata. Having a 1000 vs 1000000000 is somewhat easier to  read.

Comment (darrylmasson) [reply strax/chunk.py:39]: Valid point

**Comment by jorana** [comment]
@ ACs this is ready for merge

**Comment by jorana** [comment]
Moved the # Todo to https://github.com/AxFoundation/strax/issues/291.

I'm not sure if I agree with CodeFactor though that this makes the code better...

**Pull request merged**